src/App.tsx

```diff
@@ -15,47 +15,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE = 0,
+  ALPHABET = 1,
+  LENGTH = 2,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  isReversed: boolean,
+  sortType: SortType,
+):string[] {
+  const visible = [...goods];
+
+  visible.sort((goodFirst, goodSecond) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return goodFirst.localeCompare(goodSecond);
+      case SortType.LENGTH:
+        return goodFirst.length - goodSecond.length;
+      case SortType.NONE:
+        return 0;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visible.reverse();
+  }
+
+  return visible;
```

Comment:
you could simplify it 
```suggestion
  return isReversed ? visible.reverse() : visible;
```